### PR TITLE
Fix type signature and bugs for sparse subarray fill

### PR DESCRIPTION
**Pull request by kshyatt**
Now this `fill!` method is actually called. Used this `for` loop approach for the case when you have an element type that's a `Vector`.

### CODE REVIEW

**Comment by kshyatt** [comment]
CI failures are a timeout and something that looks unrelated on freebsd.

File: stdlib/SparseArrays/src/sparsematrix.jl [hunk 1]
```diff
@@ -2657,8 +2657,10 @@ function _spsetnz_setindex!(A::AbstractSparseMatrixCSC{Tv}, x::Tv,
                     resize!(nzvalA, nnzA)
                 end
                 r = rowidx:(rowidx+nincl-1)
-                rowvalA[r] = I
-                nzvalA[r] = x
```
Comment (dkarrasch) [inline stdlib/SparseArrays/src/sparsematrix.jl:2661]: Broadcasting doesn't work here in the case of matrix of matrices, right?

File: stdlib/SparseArrays/src/sparsematrix.jl [hunk 2]
```diff
@@ -2657,8 +2657,10 @@ function _spsetnz_setindex!(A::AbstractSparseMatrixCSC{Tv}, x::Tv,
...
+                rowvalA[r] .= I
+                for rr in r
+                    nzvalA[rr] = x
+                end
```
Comment (dkarrasch) [inline stdlib/SparseArrays/src/sparsematrix.jl:2663]: It seems like
```suggestion
                nzvalA[rr] .= Ref(x)
```
would work also in the case when `x` is a matrix. Obviously just a matter of style, not of correctness...

File: stdlib/SparseArrays/src/sparsematrix.jl (hunk 1 above)
Comment (dkarrasch) [reply stdlib/SparseArrays/src/sparsematrix.jl:2661]: > Used this for loop approach for the case when you have an element type that's a Vector.

🤦

File: stdlib/SparseArrays/src/sparsematrix.jl (hunk 2 above)
Comment (kshyatt) [reply stdlib/SparseArrays/src/sparsematrix.jl:2663]: Do you want me to change this or can this be merged?

Comment (dkarrasch) [reply stdlib/SparseArrays/src/sparsematrix.jl:2663]: As you like. I think we can merge this as is.

**Pull request merged**